### ISSUE TITLE
Duplicate kernel startup

### ISSUE THREAD

**Issue by BenRussert** [nteract/hydrogen#870]
**tl;dr**: it is possible to start up multiple kernels unintentionally while the first kernel is still getting ready

---

When a line of code is run [`createResultBubble`](https://github.com/nteract/hydrogen/blob/master/lib/main.js#L299) is called which checks to see if there is a running kernel and starts one if not (comments mine):

```js
createResultBubble(code: string, row: number, editor: atom$TextEditor) {
    if (!store.grammar) return;

    if (store.kernel) {
      // a kernel is up and running, so we can send it code to run immediately
      this._createResultBubble(store.kernel, code, row);
      return;
    }

    // no kernel is running so we need to start one up
    kernelManager.startKernelFor(store.grammar, editor, (kernel: ZMQKernel) => {
      // once the kernel is ready this callback will send it code to run
      this._createResultBubble(kernel, code, row);
    });

```

In most cases this works fine, since most kernels start up pretty quickly and the user will often wait to see output before calling more code to run.

If, however, `createResultBubble` is called two or more times in quick succession the first call must race to complete before the 2nd+ call checks for a running kernel. If the first call doesnt make it in time a second `startKernelFor` process begins.

In this gif I am pressing `shift-enter` to run and move down three times in quick succession, you can see that three different kernels begin startup:

![starts-a-bunch-o-kernels mp4](https://user-images.githubusercontent.com/10860657/27062702-6012f386-4fb2-11e7-90c2-b4f85bc9edce.gif)

**Comment by BenRussert** [comment]
As a fix would it make sense to cue commands to run then have the callback run everything in the cue? 

In most cases this would just be the code from the initial command, but if multiple inputs were entered the kernel startup process would just start once and add code to the cue if a kernel startup process had begun.

**Comment by BenRussert** [comment]
I'll make a PR later for discussion. I've got a simple fix that just prevents duplicate kernels from starting while the first is still loading up.
